lib/adapters/idb/idb.js

```diff
@@ -367,54 +367,56 @@ function init(api, opts, callback) {
     };
   };
 
-  function allDocsQuery(totalRows, opts, callback) {
```

Comment:
The diff is a little misleading. I moved a lot of logic into its own function, to avoid try/catch deoptimizations.